### PR TITLE
Added .first_or() method to Model

### PR DESCRIPTION
**Pull request by JarriqTheTechie**
Retrieves first record matching a given criteria. If no record exists matching the given criteria, a user-defined callback function will be executed.

Example usage.

```
def send_mail():
    return "mail sent"
```

`User.first_or({"name": "bob"}, lambda: send_mail())`
If no user with the name bob is found then callback function `send_mail()` will be executed.

### CODE REVIEW

**Comment by JarriqTheTechie** [comment]
@josephmancuso ready for review.

File: src/masoniteorm/models/Model.py [hunk 1]
```diff
@@ -563,6 +563,21 @@ def to_json(self):
...
 
+    @classmethod
+    def first_or(cls, wheres, callback=None, *args):
+        """Execute the query and get the first result or call a callback.
+
+        Returns:
+            Record|Mixed
+        """
+        self = cls()
+        self._check_is_callable(callback)
```
Comment (josephmancuso) [inline src/masoniteorm/models/Model.py:606]: what is this doing? You built a check for is callable but not using the return value from it anywhere

**Review by josephmancuso — CHANGES_REQUESTED** [review]

File: src/masoniteorm/models/Model.py (hunk 1 above)
Comment (josephmancuso) [reply src/masoniteorm/models/Model.py:606]: @JarriqTheTechie any update on this? This is wrong right?

**Comment by josephmancuso** [comment]
@JarriqTheTechie any update on these comments? I'd like to get this PR merged

**Pull request closed without merging**